### PR TITLE
Release/3.1.0.3

### PR DESCRIPTION
**Pull request by cklosowski**
https://github.com/awesomemotive/easy-digital-downloads/milestone/236?closed=1

| Issue # | Description |
|---------|-------------|
| #9570 | PayPal Buy now functionality no longer works for guests |
| #9569 | Exporting Orders also includes items in the Trash |
| #9558 | PayPal: Add fallback support for IPN |
| #9556 | Purchase History and Download history page shows Trashed items |
| #9553 | File Downloads report graph is formatting numbers as float and currency |
| #9550 | Many old legacy hooks running from very old EDD versions. |
| #9548 | Searching by download id on the Orders page does not work |
| #9547 | Convert UTC dates to store timezone equivalent |
| #9546 | Fix parameters for deprecated_function_run in unit tests |
| #9542 | `$download->can_purchase` condition should be revised |
| #9541 | Switching gateways when resuming a payment, assigns incorrect value to gateway column. |
| #9531 | Custom discount metadata may not be migrated to new discount object |
| #9530 | PHP Notice when viewing Customers graph |
| #9452 | Downloading files fails, get 404 messages after upgrading from to 3.0.4 |
| #9443 | edd_get_item_discount_amount assumes discount options have price IDs |
| #9414 | Logging in from the {receipt_link} page simply refreshes the login form |

EDD Blocks 2.0.2
https://github.com/awesomemotive/edd-blocks/milestone/9?closed=1

| Issue # | Description |
|---------|-------------|
| [#131](https://github.com/awesomemotive/edd-blocks/issues/131) | Update receipt redirect when showing the login form |
| [#125](https://github.com/awesomemotive/edd-blocks/issues/125) | Checkout block fails when no first name explicitly set in user account |
| [#123](https://github.com/awesomemotive/edd-blocks/issues/123) | Do not show 'Trash' orders in order history |
| [#121](https://github.com/awesomemotive/edd-blocks/issues/121) | Requiring Login fatal errors with specific form options |

### CODE REVIEW

**Comment by mihaijoldis** [comment]
For the item in the changelog with the trashed orders showing in history. can we add at the end of the line in the changelog something like (edd-purchase-history.php updated) or something.

Just to reflect that a core theme file has been updated so that perhaps users can easily notice that in case they have update theme files, and also for theme developers

`note to make a reference that the download-history template has also had a modification so I can remember to add it?`

File: readme.txt
Comment (robincornett) [inline readme.txt:243]:
```suggestion
* Fix: PayPal Buy Now functionality has been fixed for guest users.
```

File: readme.txt
Comment (robincornett) [inline readme.txt:248]:
```suggestion
* Fix: In some timezone settings, the reports could incorrectly display.
```

File: readme.txt
Comment (robincornett) [inline readme.txt:246]:
```suggestion
* Fix: The file downloads graph was formatting integers as currency.
```

File: readme.txt
Comment (robincornett) [inline readme.txt:245]:
```suggestion
* Fix: Trashed orders were showing in the purchase history shortcode (templates updated: history-downloads.php and history-purchases.php).
```

**Review by robincornett — CHANGES_REQUESTED** [review]

**Review by robincornett — APPROVED** [review]

**Comment by cklosowski** [comment]
Last build test failed, but the one before that passed, going to merge this once buildkite settles down :)

**Pull request merged**
